cmd/README.md

```diff
@@ -0,0 +1,738 @@
+
+``` go
+/* General guidelines for precompile development:
+1- Read the comment and set a suitable contract address in precompile/params.go. E.g:
+	HelloWorldAddress = common.HexToAddress("ASUITABLEHEXADDRESS")
+2- Set gas costs here
+3- It is recommended to only modify code in the highlighted areas marked with "CUSTOM CODE STARTS HERE". Modifying code outside of these areas should be done with caution and with a deep understanding of how these changes may impact the EVM.
+Typically, custom codes are required in only those areas.
+4- Add your upgradable config in params/precompile_config.go
+5- Add your precompile upgrade in params/config.go
+6- Add your solidity interface and test contract to contract-examples/contracts
+7- Write solidity tests for your precompile in contract-examples/test
+8- Create your genesis with your precompile enabled in tests/e2e/genesis/
+9- Create e2e test for your solidity test in tests/e2e/solidity/suites.go
+10- Run your e2e precompile Solidity tests with 'E2E=true ./scripts/run.sh'
+*/
+```
+
+## Step 1: Set Contract Address
+
+In `./precompile/params.go` let's set a precompile address. We can cut the
+address from the var declaration block and remove it from the precompile. 
+
+![](2022-09-01-22-46-00.png)
+![](2022-08-24-16-45-48.png)
+
+Now when subnet-evm sees the `HelloWorldAddress` as input when executing [`CALL`](../core/vm/evm.go#L222), [`STATICCALL`](../core/vm/evm.go#L401), [`DELEGATECALL`](core/vm/evm.go#L362), [`CALLCODE`](core/vm/evm.go#L311), it can [run the precompile](https://github.com/ava-labs/subnet-evm/blob/master/core/vm/evm.go#L271-L272) if the precompile is enabled.
+
+## Step 2: Set Gas Costs
+
+Set up gas costs. In `precompile/params.go` we have `writeGasCostPerSlot` and `readGasCostPerSlot`. This is a good starting point for estimating gas costs. 
+
+``` go
+// Gas costs for stateful precompiles
+const (
+	writeGasCostPerSlot = 20_000
+	readGasCostPerSlot  = 5_000
+)
+```
+
+For example, we will be setting and getting our greeting in one slot so we can define the gas costs as follows. 
+
+``` go
+	SayHelloGasCost uint64    = 20_000 
+	SetGreetingGasGost uint64 = 5000
+```
+
+
+**Example:** 
+The sha256 precompile computes gas with the following equation
+``` go
+// This method does not require any overflow checking as the input size gas costs
+// required for anything significant is so high it's impossible to pay for.
+func (c *sha256hash) RequiredGas(input []byte) uint64 {
+	return uint64(len(input)+31)/32*params.Sha256PerWordGas + params.Sha256BaseGas
+}
+```
+
+## Step 3: Add Custom Code
+
+Ok time to `CTRL F` throughout the file with `CUSTOM CODE STARTS HERE` to find the areas in the precompile that we need to modify. 
+
+![](2022-09-01-22-48-26.png)
+
+If we use any of the following imports we can remove the reference imports.
+We can come back if needed to remove. Next we see this in `Equals()`.
+
```

Comment:
If the imports are unused then we can remove this line as well as the import itself.

We include this in the template so that the template can simply include all of the imports that it might need and avoid generating a file that may have un-needed imports which would result in a compiler warning.

Reply:
thanks